src/notification.ts

```diff
@@ -0,0 +1,272 @@
+
+      return this.apiClient.post('/v2/notifications/email',
+        createNotificationPayload('email', templateId, emailAddress, personalisation, reference, emailReplyToId));
+    }
+
+    /**
+     *
+     * @param {String} templateId
+     * @param {String} phoneNumber
+     * @param {Object} options
+     *
+     * @returns {Promise}
+     */
+    public sendSms(templateId: string, phoneNumber: string, options?: any): any {
+      options = options || {};
+      const err = checkOptionsKeys(['personalisation', 'reference', 'smsSenderId'], options);
+      if (err) {
+        return Promise.reject(err);
+      }
+
+      const personalisation = options.personalisation || undefined;
+      const reference = options.reference || undefined;
+      const smsSenderId = options.smsSenderId || undefined;
+
+      return this.apiClient.post('/v2/notifications/sms',
+        createNotificationPayload('sms', templateId, phoneNumber, personalisation, reference, smsSenderId));
+    }
+
+    /**
+     *
+     * @param {String} notificationId
+     *
+     * @returns {Promise}
+     */
+    public getNotificationById(notificationId: string): RequestPromise {
+      return this.apiClient.get('/v2/notifications/' + notificationId);
+    }
+
+    /**
+     *
+     * @param {Object} An optional object literal that filters the query. Accepts the following filters:
+     *    @param {String} templateType
+     *    @param {String} status
+     *    @param {String} reference
+     *    @param {String} olderThanId
+     *
+     * @returns {Promise}
+     *
+     */
+    public getNotifications(options?: { templateType?: string, status?: string, reference?: string, olderThanId?: string }): RequestPromise {
+      return this.apiClient.get('/v2/notifications' + buildGetAllNotificationsQuery(options));
+    }
+
+    /**
+     *
+     * @param {String} templateId
+     *
+     * @returns {Promise}
+     */
+    public getTemplateById(templateId: string): RequestPromise {
+      return this.apiClient.get('/v2/template/' + templateId);
+    }
+
+    /**
+     *
+     * @param {String} templateId
+     * @param {Integer} version
+     *
+     * @returns {Promise}
+     */
+    public getTemplateByIdAndVersion(templateId, version): RequestPromise {
+      return this.apiClient.get('/v2/template/' + templateId + '/version/' + version);
+    }
+
+    /**
+     *
+     * @param {String} templateType
+     *
+     * @returns {Promise}
+     */
+    public getAllTemplates(templateType?: string): RequestPromise {
+        let templateQuery = '';
+
+        if (templateType) {
+          templateQuery = '?type=' + templateType;
+        }
+
+        return this.apiClient.get('/v2/templates' + templateQuery);
+    }
+
+    /**
+     *
+     * @param {String} templateId
+     * @param {Dictionary} personalisation
+     *
+     * @returns {Promise}
+     */
+    public previewTemplateById(templateId: string, personalisation?: any): RequestPromise {
```

Comment:
Type of `any`?

Reply:
Return type? Isn't it better to be specific?

Reply:
Sorry I meant that `personalisation` has type `any` but it should be a map/object/dict type, no?